documentation/guides/clock_systems.md

```diff
@@ -0,0 +1,275 @@
+ struct
+ {
+   UsbClockSelect clock = UsbClockSelect::kSystemClock;
+   UsbDivider divider   = UsbDivider::kDivideBy1;
+ } usb = {};
+
+ /// Represents the configuration options for SPIFI
+ struct
+ {
+   SpifiClockSelect clock = SpifiClockSelect::kSystemClock;
+   uint8_t divider        = 1;
+ } spifi = {};
+
+ /// Defines the peripheral clock divider amount
+ uint8_t peripheral_divider = 1;
+ /// Defines the EMC clock divider amount
+ EmcDivider emc_divider = EmcDivider::kSameSpeedAsCpu;
+};
+```
+
+ClockConfiguration objects are POD (plain 'ol data), thus they will not have
+methods or even static functions to do certain operations, although this may
+change in the future.
+
+Each field in the structure that is not an number or bool is an enumeration used
+to constrain the possible options a particular field can be.
+
+!!! Note
+    In most cases, the clock tree diagram in the user manual for the platform
+    will need to be consulted in order to understand how each of the fields in
+    the ClockConfiguration object connects with each other. If an field is
+    missing and you need it for your project, please be sure to add a feature
+    request issue on the github.
+
+### Clock Configuration Retrieval & Modification
+
+Here we grab the platform controller and store it into a `system` reference
+variable. Next we use that reference to get the `ClockConfiguration` using
+`GetClockConfiguration()`. This is version of `GetClockConfiguration()` takes
+the type of the `ClockConfiguration` object which is used to cast and return the
+a reference to that `ClockConfiguration` type rather than `void*`. Notice that
+the platform configuration object is typically defined within the
+`SystemController`.
+
+```C++
+auto & system = sjsu::SystemController::GetPlatformController();
+auto & config = system.GetClockConfiguration<
+   sjsu::lpc40xx::SystemController::ClockConfiguration>();
+```
+
+Now that we have a reference to the config object, we can modify it.
+
+```C++
+// Select system clock (use crystal oscillator directly)
+config.cpu.clock =
+   sjsu::lpc40xx::SystemController::CpuClockSelect::kSystemClock;
+// Make sure to utilize the internal oscillator only.
+config.system_oscillator =
+   sjsu::lpc40xx::SystemController::OscillatorSource::kIrc;
+// Make sure PLL0 is enabled and available
+config.pll[0].enabled  = true;
+config.pll[0].multiply = 4;
+```
+
+Finally, in order to actually use these new settings, we have two methods:
+
+#### Method 1
+
+```C++
+auto result = system.Initialize();
+if (!result)
+{
+  // Handle error ...
+}
+```
+
+This option seems like the obvious choice but has a fatal issue. During platform
+initialization, peripherals and other drivers that are enabled at startup will
+have adjusted their own clock frequencies based on the frequency they were fed
+initially. If you change these without also rerunning the those driver's
+initialize methods, their clocks will be incorrect and they will not performed
```

Comment:
initialize methods, their clocks will be incorrect and ~~they~~ **the drivers** will not **perform~~ed~~**